src/modules/proofs/services/ProofService.ts

```diff
@@ -395,6 +398,12 @@ export class ProofService {
       )
     }
 
+    // Get the matching attachments to the requested credentials
+    const attachments = await this.getRequestedAttachmentsForRequestedCredentials(
+      indyProofRequest,
+      requestedCredentials
+    )
```

Comment:
Would it be possible to check beforehand whether this concerns linked attachments? There are not used that commonly, but this method always retrieves records for proofs. Would be nice to know beforehand if the retrieval of records for attachments is necessary

Reply:
Unfortunately, it can only be known from the `credential attribute value`. It does indeed feel quite inefficient, but the only alternative I could think is manually sending an attachment object when you call `createPresentation`.